### DISCUSSION TITLE
BooleanGroup required validation not obeyed on action

### DISCUSSION THREAD

**Discussion by bkuhl** [laravel/nova-issues#5386] [Q&A] [answered]
- Laravel Version: 8.52.4
- Nova Version: 4.22.1
- PHP Version: 8.1.2
- Database Driver & Version: MySQL 8.0.3
- Operating System and Version: OSx 13.0.1 (22A400)
- Browser type and version: Chrome 108.0.5359.124 (Official Build) (arm64)
- Reproduction Repository: https://github.com/###/###

### Description:
Defining a `BooleanGroup` on an action does not obey `required` validation rule.  This could be because there's not an attribute on the model that correlates to this field, however it seems like this should still support collecting simple feedback for an action, similar to `Select`.

### Detailed steps to reproduce the issue on a fresh Nova installation:
 - Create an action for a resource
 - Specify the following field:
 ```
     public function fields(NovaRequest $request)
    {
        return [
            BooleanGroup::make('Some Options')
                ->options([
                    'some' => 'value',
                ])
                ->required()
                ->rules('required'),
      ]
   }
 ```
 - Do not select the checkbox and run the action and notice the form validation is ignored and the action executes

**Comment by crynobone** [comment]
By default, given the above example, `BooleanGroup` field would submit the following:

```json
{ "some": false }
```

So, using just `required` wouldn't make any different.

**Comment by bkuhl** [comment]
Can you help explain to me why any value would be submitted when no checkbox was checked?

**Comment by bkuhl** [comment]
I should elaborate... this is a simplified example.  In my case, the value of the checkbox was a string, however the action executed without anything being checked.

**Comment by crynobone** [comment]
`BooleanGroup` only return a set a boolean collections. E.g:

```php
BooleanGroup::make('Permissions')->options([
    'create' => 'Create',
    'read' => 'Read',
    'update' => 'Update',
    'delete' => 'Delete',
 ])
```

would return `object{create: bool, read: bool, update: bool, delete: bool}`

**Comment by davidhemphill** [comment, marked as answer]
@bkuhl The field starts with a valid state by its nature. The required validation in this case just checks the value is passed in the request. 

It sounds like you're trying to create a list of "are you sure you did this?" confirmations before running the action, but the user is allowed to ignore the list before submitting. Is that correct? If so, you could accomplish that like this:

```php
BooleanGroup::make('Everything looks correct?')->options([
    'something' => 'Something',
    'else' => 'Something Else',
])->rules([
    function (string $attribute, mixed $value, Closure $fail) {
        $options = json_decode($value, true);
        $confirmedCount = count(array_filter($options, fn($val) => $val));

        if ($confirmedCount !== count($options)) {
            $fail('Nope, ya gotta check everything!');
        }
    }
]),
```

**Comment by bkuhl** [comment]
Ok, this all makes a lot more sense now, thank you.  I'm actually just looking to have checkboxes for each relation to perform the action with.
